app/src/main/java/org/elasticsearch/node/internal/CrateSettingsPreparer.java

```diff
@@ -155,12 +155,14 @@ private static String randomNodeName() {
                 while (line != null) {
                     String[] fields = line.split("\t");
                     if (fields.length == 0) {
+                        input.close();
                         throw new RuntimeException("Failed to parse the names.txt. Malformed record: " + line);
                     }
                     names.add(fields[0]);
                     line = reader.readLine();
                 }
             }
+            input.close();
```

Comment:
doesn't this mean it's not closed if an IOException is thrown?

In other places `IOUtils.closeWhileHandlingException` is used in the finally block for that reason